apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -3198,27 +3203,32 @@ handle_mining_h2(Req, Pid) ->
 	Peer = ar_http_util:arweave_peer(Req),
 	case read_complete_body(Req, Pid) of
 		{ok, Body, Req2} ->
-			case {ar_pool:is_client(), ar_coordination:is_exit_peer()} of
-				{true, true} ->
-					PoolPeer = ar_pool:pool_peer(),
-					spawn(fun() ->
-						ar_http_iface_client:cm_h2_send(PoolPeer, Body) end);
-				_ ->
-					case ar_serialize:json_decode(Body, [{return_maps, true}]) of
-						{ok, JSON} ->
-							case catch ar_serialize:json_map_to_candidate(JSON) of
-								{'EXIT', _} ->
-									{400, #{}, jiffy:encode(#{ error => invalid_json }), Req2};
-								Candidate ->
-									?LOG_INFO([{event, h2_received},
-											{peer, ar_util:format_peer(Peer)}]),
+			case ar_serialize:json_decode(Body, [{return_maps, true}]) of
+				{ok, JSON} ->
+					case catch ar_serialize:json_map_to_candidate(JSON) of
+						{'EXIT', _} ->
+							{400, #{}, jiffy:encode(#{ error => invalid_json }), Req2};
+						Candidate ->
+							?LOG_INFO([{event, h2_received},
+									{peer, ar_util:format_peer(Peer)}]),
+							case {ar_pool:is_client(), ar_coordination:is_exit_peer()} of
+								{true, true} ->
+									PoolPeer = ar_pool:pool_peer(),
+									Jobs = #pool_cm_jobs{ h1_read_jobs = [Candidate] },
```

Comment:
`h1_read_jobs` means we've calculated an H2 solution be we don't have the h1 chunk locally so we're submitting a job for a different peer to "read the h1 chunk", right?

I'm trying to sort out why we submit an `h1` related job after mining an `h2` solution.

Reply:
Yes, except for we do not look for the first chunk at the second peer.

Reply:
got it... I think. I can sort it out later. Resolving!